src/models/model.cpp

```diff
@@ -1920,6 +1942,12 @@ void model::replace_layer(OwningLayerPtr&& new_layer, std::string const& old_lay
   parent.replace_child_layer(new_layer, parent.find_child_layer_index(l));
   new_layer->add_parent_layer(m_layers[old_layer_index-1]);
 
+  // Remove weights for the old layer
+  auto old_weights_ptrs = m_layers[old_layer_index]->get_weights_pointers();
+  for (const auto w : old_weights_ptrs) {
+     this->remove_weights(std::shared_ptr<weights>(w)->get_name());
+  }
```

Comment:
We should mention this behavior in the documentation.

Reply:
Added a comment for both